test/fluree/db/ledger/docs/identity/signatures_test.clj

```diff
@@ -0,0 +1,204 @@
+                        "history"     {:history [["_collection/name" "_user"]]}
+                        "multi-query" {:person {:select ["*"] :from "person"}}}
+          opts         {:throw-exceptions false}
+          resps        (reduce-kv (fn [acc key val]
+                                    (conj acc (send-parse-request ledger key val opts)))
+                                  [] endpoint-map)]
+
+      (is (every? #(= 401 (:status %)) resps))
+
+      (is (every? #(= "Request requires authentication." (:message %)) resps)))))
+
+
+;; Endpoint test - /query - should return filtered results
+
+(deftest signed-queries-test
+  (let [ledger           (test/rand-ledger test/ledger-chat)
+
+        [keys] (create-auths ledger)
+
+        _                (test/transact-schema ledger "chat.edn" :clj)
+        _                (test/transact-schema ledger "chat-preds.edn" :clj)
+        _                (test/transact-data ledger "chat.edn" :clj)
+
+        req              {:headers {"content-type" "application/json"}
+                          :body    (json/stringify {:select ["*"] :from "person"})}
+
+        q-endpoint       (str endpoint ledger "/query")
+
+        root-auth        (get-in keys [0 :private])
+        person-auth      (get-in keys [1 :private])
+        no-handle-auth   (get-in keys [2 :private])
+
+        signed-root      (http-signatures/sign-request :post q-endpoint req
+                                                       root-auth)
+        signed-person    (http-signatures/sign-request :post q-endpoint req
+                                                       person-auth)
+        signed-no-handle (http-signatures/sign-request :post q-endpoint req
+                                                       no-handle-auth)
+
+        {root-body :body :as root-resp}
+        (-> signed-root
+            (->> (http/post q-endpoint))
+            deref
+            (test/safe-update :body string? json/parse))
+
+        {person-body :body :as person-resp}
+        (-> signed-person
+            (->> (http/post q-endpoint))
+            deref
+            (test/safe-update :body string? json/parse))
+
+        {no-handle-body :body :as no-handle-resp}
+        (-> signed-no-handle
+            (->> (http/post q-endpoint))
+            deref
+            (test/safe-update :body string? json/parse))
+
+        responses        [root-resp person-resp no-handle-resp]
+        bodies           [root-body person-body no-handle-body]]
+
+    (is (every? #(= 200 %) (map :status responses)))
+
+    (is (apply = (map count bodies)))
+
+    (is (some :person/handle root-body))
+
+    (is (some :person/handle person-body))
+
+    (is (not-any? :person/handle no-handle-body))))
+
+
+;; Endpoint test - /transact - should throw error, /command endpoint shouldn't
+
+(deftest transact-endpoint-test
+  (testing "The /transact endpoint should not work.")
```

Comment:
The following `let` form should be within this `testing` form so the error messages line up in case the test fails.